### DISCUSSION TITLE
Tab UI

### DISCUSSION THREAD

**Discussion by trevarj** [osa1/tiny#284] [Ideas]
I'm excited to be the first person to make a discussion 😃 

I've been thinking / hacking on some ideas for a new tab user interface for a while now. Finally had time and energy to sit down and start working on something that I think will be nice, while still being minimal (and tiny), and also not removing the current functionality.

At first, I really liked the idea of a wrapping tab bar (#125), but after some thought I decided that this would not be my ideal alternative to the single line, scrolling tab area. I think that a better option is the more traditional "tab panel" that you would see in a GUI program, like Hexchat:

![Tab Panel](https://user-images.githubusercontent.com/5448324/101979347-79f2f700-3c6d-11eb-873c-fe4d15fab302.png)

Obviously, this is really primitive at the moment, but it's something that I could really see myself using more effectively. Right now I'm actually afraid of adding/keeping channels due to the tab line becoming too large! 😖 Being able to scroll this list, as done with the message area (pgup/pgdown), combined with popping active channels to the top of the list would be really cool.

Of course, I'm open to any ideas/preferences that others have.

Side note, this could be the start of having a user list too. Maybe split the channel list in half horizontally and list users there, instead of adding another panel to the right side.

**Comment by osa1** [comment]
Wow, this looks great @trevarj , thanks for doing this!

I agree that this may also pave the way for the nick list.

Just one suggestion, I wonder if instead showing channels and nicks of a server at the same column with server, it would be easier to read if we indent a server's tabs with one space. So instead of

```
freenode
##rust
oftc
#tiny
gitter
#rust-lang
```

We do

```
freenode
 ##rust
oftc
 #tiny
gitter
 #rust-lang
```

**Reply by trevarj** [reply]
Yes! Was thinking the same. A nice way to differentiate servers and channels.

**Comment by trevarj** [comment]
Tracking branch: https://github.com/trevarj/tiny/tree/tab-area

**Comment by trevarj** [comment]
Demo of what the page arrows look like:

![image](https://user-images.githubusercontent.com/5448324/102090313-6216a100-3e2e-11eb-8fb0-e281fce84471.png)
